### DISCUSSION TITLE
Add a delay option for the "Window Throw" feature

### DISCUSSION THREAD

**Discussion by TheBillyZee** [rxhanson/RectanglePro-Community#320] [Ideas]
I'm currently using the middle mouse button on the MX Master 3 as the window throw button. It works well, except when I middle-click on a link, like in this case,

![08-05-2023 at 00 23 34](https://github.com/rxhanson/RectanglePro-Community/assets/33535317/b82e0467-9101-42f5-bd82-815f06acb52d)

The mouse gets stuck and the only way to unstuck it is by when I open raycast.

To address this issue, I suggest adding an option to introduce a delay. The hookshot will only activate once the specified delay time has passed.

![08-04-2023 at 23 51 09@2x](https://github.com/rxhanson/RectanglePro-Community/assets/33535317/08c9484a-803f-4dcf-8cfc-31c699ff9d4e)

Another fix might be to only have the Hookshot only activate when the cursor is on the app top bar, something similar to what Swish does for trackpads.

**Comment by rxhanson** [comment]
Thanks for the details. My recommendation here is to give it a small "Min Dist to Appear". If that doesn't work for you or that's undesirable, let me know.

**Reply by TheBillyZee** [reply]
Increasing the "Min Dist to Appear" setting still causes the middle click to get stuck when the Middle Click button is quickly pressed and released. However, holding it down does not cause any issues.

![08-05-2023 at 20 47 58](https://github.com/rxhanson/RectanglePro-Community/assets/33535317/a7e07e75-8d1d-4475-947c-4bc919662704)
